### ISSUE TITLE
JSX support?

### ISSUE THREAD

**Issue by neemzy** [Kegsay/flow-jsdoc#1]
Hi!

When running the tool on a React-powered codebase, I get the (predictable) following error:

```
/media/neemzy/storage/aecf/gui • node_modules/.bin/flow-jsdoc -d ./app/Resources/scripts/ -o ./annotated

/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:5702
            throw e;
            ^
Error: Line 1481: Unexpected token <
    at constructError (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:2407:21)
    at createError (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:2426:17)
    at unexpectedTokenError (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:2500:13)
    at throwUnexpectedToken (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:2505:15)
    at parsePrimaryExpression (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:3307:13)
    at inheritCoverGrammar (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:2681:18)
    at parseLeftHandSideExpressionAllowCall (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:3414:20)
    at inheritCoverGrammar (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:2681:18)
    at parsePostfixExpression (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:3490:16)
    at parseUnaryExpression (/media/neemzy/storage/aecf/gui/node_modules/flow-jsdoc/node_modules/esprima/esprima.js:3550:20)
```

I'm unfamiliar with Esprima and code parsing in general; is there any way to make the parsing JSX-friendly in order to avoid this?

Thanks!

**Comment by Kegsay** [comment]
Yeah, this doesn't do JSX yet. I would like to support JSX. I haven't looked into what would be required to make Esprima able to parse it. As a horrendous workaround, you *may* be able to transpile the React code *then* add flow annotations.

Transpiling of raw React code is very simple as it just does a blunt mapping of:
```
<a href="someplace">Something</a>
```
to:
```
React.createElement("a", {href:"someplace"}, "Something");
```

So if you just did that through babel, and then applied `flow-jsdoc`, you should get an output which Flow will happily parse.

**Comment by Kegsay** [comment]
Looks like Esprima does now support this (as of late-August) - https://github.com/jquery/esprima/issues/1527

**Comment by neemzy** [comment]
Thanks for the info @Kegsay, I'll be looking into this when I get the time. Cheers!

**Comment by Kegsay** [comment]
Done. JSX support has landed in `flow-jsdoc@0.2.0`.
